### PR TITLE
[Misc]Add pyproject.toml for autopep8 & black.

### PR DESCRIPTION
**Pull request by frozenbugs**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by jermainewang** [comment]
What's the relation of this PR v.s. #4544 ? Are they the same tools?

**Comment by jermainewang** [comment]
Just realized this is for autopep8. No concern from me.

**Comment by jermainewang** [comment]
I've tried it on `python/dgl/heterograph.py`. It moderately expands the file from 6464L -> 6723L and most of the changes are striding function arguments with newlines. In general, it is a pretty safe tool to use.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[tool.autopep8]
+
+max_line_length = 80
+in-place = true
+aggressive = 3
+# Add the path to here if you want to exclude them from autopep8 auto reformat.
+exclude = '''
```
Comment (jermainewang) [inline pyproject.toml:7]: The following folders should also be excluded:
- apps, benchmarks, cmake, conda, docker, docs, featgraph, third_party, tutorials

The following folders can be omitted for now but should be checked later:
- tests

**Review by jermainewang — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (frozenbugs) [reply pyproject.toml:10]: Added part of it. It is not clear to me why python code in apps/ should be ignored.
If we feel it is important to add, we can do it later.

Comment (jermainewang) [reply pyproject.toml:15]: They are legacy codes that shall we removed.
* dgl-ke has been moved to https://github.com/awslabs/dgl-ke
* dgl-lifesci has been moved to https://github.com/awslabs/dgl-lifesci

**Pull request merged**